packages/angular_devkit/build_angular/src/tools/esbuild/bundler-context.ts

```diff
@@ -169,8 +169,8 @@ export class BundlerContext {
       outputFile.path = relativeFilePath;
 
       if (entryPoint) {
-        // The first part of the filename is the name of file (e.g., "polyfills" for "polyfills.7S5G3MDY.js")
-        const name = basename(relativeFilePath).split('.', 1)[0];
+        // The first part of the filename is the name of file (e.g., "polyfills" for "polyfills-7S5G3MDY.js")
+        const name = basename(relativeFilePath).split('-', 1)[0];
```

Comment:
This line is causing the problem as the extension is no longer removed. Also in the case using slice can cause problems for entrypoints name with a dashed Example `material-styles.css`. I propose to you a regexp.
```suggestion
        const name = basename(relativeFilePath).replace(/(?:-[\dA-Z]{8})?\.[a-z]{2,3}$/, '');
```

Reply:
Thanks Alan, let's try that